tools/codegen/gen.py

```diff
@@ -205,12 +206,11 @@ def cpp_string(s: str) -> str:
 # to be generated.  This pattern makes it convenient to use map, concatMap
 # and similar functional combinators.
 
-def static_dispatch_keys(backend: Optional[BackendIndex]) -> List[DispatchKey]:
-    if backend is None:
+def static_dispatch_keys(backends: List[BackendIndex]) -> List[DispatchKey]:
+    if backends is None:
```

Comment:
nit: you can probably remove the none check now that the input isn't `Optional` (passing in `None` should fail mypy)